main.cpp

```diff
@@ -19,13 +19,39 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : myrewriter(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    
+       		if (const auto *expr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+			auto type = Lexer::getSourceText(
+			CharSourceRange::getTokenRange(
+			expr->getLParenLoc().getLocWithOffset(1),
+                                    expr->getRParenLoc().getLocWithOffset(-1)),
+                                *Result.SourceManager, LangOptions());
+
+     auto cast_str = ("static_cast<"+type+">").str();
+
+	if(!isa<ParenExpr>(expr->getSubExprAsWritten())) // if no parenteses we need to add them
+	{
+		cast_str=("static_cast<"+type+">(").str(); // add left bracket
+	
+ 		myrewriter.InsertText(
+                        Lexer::getLocForEndOfToken(
+                            expr->getSubExprAsWritten()->IgnoreImpCasts()->getEndLoc(),

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
getSubExprAsWritten() retrieves the cast subexpression as it was written in the source code, looking through any implicit casts or other intermediate nodes introduced by semantic analysis.
getSubExpr() retrieves the first cast subexpression and IgnoreImpCasts() skips past any implicit casts which might surround this expression until reaching a fixed point.  

So it was no sense to use getSubExprAsWritten() and IgnoreImpCasts(), we can use just getSubExprAsWritten(). 

But we must use IgnoreImpCasts() with getSubExpr() cause if we will write only getSubExpr()  we can get unexpected result. 

getSubExpr()->IgnoreImpCasts() and getSubExprAsWritten() will give the same result.